pkg/scheduler/cache/event_handlers.go

```diff
@@ -145,7 +145,7 @@ func (sc *SchedulerCache) deleteTask(pi *kbapi.TaskInfo) error {
 
 	if len(pi.NodeName) != 0 {
 		node := sc.Nodes[pi.NodeName]
-		if node != nil {
+		if node != nil && !isTerminated(pi.Status) {
```

Comment:
what happen if task finished?

Reply:
I used the nginx images in containers. It should never exit as expected. Currently I do not know why it exit and the pod enter into Succeeded phase. I will use another kind of image to simulate the complete successfully.

Reply:
Had a test. 
If the task/pod complete successfully, there is not `DeletePod` event. All thing goes well.
If the task/pod is evicted, there is `DeletePod` event. However the task state is `DeletePod/UpdatePod` is releasing. All thing goes well too.
Only with nginx, the Succeded state happens and leads to this issue.

Reply:
@k82cn , please help to review and approve. Thanks.

Reply:
why 82 commits?

Reply:
The 82 commits includes many other member's commits after using 'pull --rebase'.

Reply:
@k82cn any more comments ?

Reply:
please remove other commits.

Reply:
@k82cn Other comments is removed.

Reply:
still 22 commits for 3 files?

Reply:
> If the task/pod is evicted, there is DeletePod event. However the task state in DeletePod/UpdatePod is releasing. 

What's happen if no grace period?

Reply:
There is no issue if no grace perod.

Reply:
> still 22 commits for 3 files?
The redundant commit is removed.

Reply:
> There is no issue if no grace perod.

Why?

Reply:
If grace period is set 0, the pod is deleted inmediately and only has DeletePod event.  So the task must be on the node when calling DeletePod func.
If grace period is not set 0, there are two events including UpdatePod and DeletePod. If the task is removed from node in UpdatePod func, the task does not exist on node any more when calling DeletePod func.